src/frontend/src/modules/Transactions/FilterTransaction.vue

```diff
@@ -322,6 +340,71 @@ export default {
         this.filter.to = toDate.toISOString().split("T")[0] + " 23:59:59"
       }
     },
+    async loadMiniGrids() {
+      try {
+        await this.miniGridService.getMiniGrids()
+      } catch (error) {
+        console.error("Failed to load mini grids:", error)
+      }
+    },
+
+    async loadCities() {
+      try {
+        await this.cityService.getCities()
+      } catch (error) {
+        console.error("Failed to load cities:", error)
+      }
+    },
+
+    async loadCurrencyList() {
+      try {
+        await this.currencyListService.getCurrencyList()
+      } catch (error) {
+        console.error("Failed to load currency list:", error)
+      }
+    },
+
+    async exportCustomers() {
+      try {
+        const data = {
+          format: this.exportFilters.format,
+        }
+
+        // Add optional filters if they are set
+        if (this.exportFilters.isActive !== null) {
+          data.isActive = this.exportFilters.isActive
+        }
+        if (this.exportFilters.miniGrid) {
+          data.miniGrid = this.exportFilters.miniGrid
+        }
+        if (this.exportFilters.village) {
+          data.village = this.exportFilters.village
+        }
+        if (this.exportFilters.deviceType) {
+          data.deviceType = this.exportFilters.deviceType
+        }
+        if (this.selectedAgentId) {
+          data.agent = this.selectedAgentId
+        }
+
+        const response = await this.customerExportService.exportCustomers(data)
+        const blob = new Blob([response.data])
+        const downloadUrl = window.URL.createObjectURL(blob)
+        const a = document.createElement("a")
+        a.href = downloadUrl
+        const contentDisposition = response.headers["content-disposition"]
+        const fileNameMatch = contentDisposition?.match(/filename="(.+)"/)
+        a.download = fileNameMatch ? fileNameMatch[1] : "export_customers.csv"
```

Comment:
You are defaulting to only csv here I see. Don't we need to handle the excel export type as well?